src/pybind/mgr/snap_schedule/module.py

```diff
@@ -0,0 +1,155 @@
+"""
+Copyright (C) 2019 SUSE
+
+LGPL2.1.  See file COPYING.
+"""
+import errno
+import rados
+from contextlib import contextmanager
+from mgr_module import MgrModule, CLIReadCommand, CLIWriteCommand
+from threading import Event
+try:
+    import queue as Queue
+except ImportError:
+    import Queue
+
+SNAP_SCHEDULE_NAMESPACE = 'cephfs-snap-schedule'
+DEFAULT_FS = 'cephfs'
+
+
+class Module(MgrModule):
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self._initialized = Event()
+
+        self._background_jobs = Queue.Queue()
+
+    def serve(self):
+        self._initialized.set()
+
+    def handle_command(self, inbuf, cmd):
+        self._initialized.wait()
+        return -errno.EINVAL, "", "Unknown command"
+
+    @CLIReadCommand('fs snap-schedule ls',
+                    'name=path,type=CephString,req=false '
+                    'name=subvol,type=CephString,req=false '
+                    'name=fs,type=CephString,req=false',
+                    'List current snapshot schedules')
+    def snap_schedule_ls(self, path=None, subvol=None, fs=DEFAULT_FS):
+        raise NotImplementedError
+
+    @CLIReadCommand('fs snap-schedule get',
+                    'name=path,type=CephString '
```

Comment:
Is this path relative to the vol/subvol?

Reply:
Yes, if a subvolume is passed. Iiuc that is how [sub]volumes are currently implemented. I suppose they could also imply a specific fs instance to be used.